<p>The use of wrapper objects for primitive types is gratuitous, confusing and dangerous. Simple literals should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var x = new Boolean(false);
if (x) {
  alert('hi');  // Shows 'hi'.
}
</pre>
<h2>Compliant Solution</h2>
<pre>
var x = false;
if (x) {
  alert('hi');
}
</pre>

