<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h2>Noncompliant Code Example</h2>
<p>With a maximum number of 4 parameters:</p>
<pre>
function doSomething(param1, param2, param3, param4, param5) {
...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function doSomething(param1, param2, param3, param4) {
...
}
</pre>
<h2>Exceptions</h2>
<p>The rule ignores constructors where parameters are <strong>all</strong> parameter properties:</p>
<pre>
class C {
  constructor(
    param1: number,
    param2: boolean,
    param3: string,
    param4: string[],
    param5: number | string
  ) {}
}
</pre>

