<p>For consistency single-quotes (') are preferred to double-quotes ("). This is helpful when creating strings that include HTML. </p>
<h2>Noncompliant Code Example</h2>
<pre>
var firstParameter = "something";
</pre>
<h2>Compliant Solution</h2>
<pre>
var firstParameter = 'something';
</pre>
<h2>Exceptions</h2>
<p>Strings that contain quotes are ignored.</p>
<pre>
let sheSaid = '"Whatever!" she replied.'  // ignored
</pre>

