<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:</p>
<pre>
public class myClass {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class MyClass {...}
</pre>

