<p>In addition to being obtuse from a syntax perspective, function constructors are also dangerous: their execution evaluates the constructor's string
arguments similar to the way <code>eval</code> works, which could expose your program to random, unintended code.</p>
<p>Instead, simply declare the function as usual.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var greeting = new Function("name", "return 'Hello ' + name");  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
var greeting = function (name) {
  return 'Hello ' + name;
}
</pre>
<p>or</p>
<pre>
function greeting (name) {
  return 'Hello ' + name;
}
</pre>

