<p>
  JavaScript is flexible enough to allow you to store values in an array with either numeric or named indexes. That is, it supports associative arrays. But creating and populating an object in JavaScript is just as easy as an array, and more reliable if you need named members.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
let arr = [];
arr[0] = 'a';
arr['name'] = 'bob';  // Noncompliant
arr[1] = 'foo';

for (let i = 0; i < arr.length; i++) {
  writeln(arr[i]);  // outputs: a, foo. bob omitted
}
</pre>

<h2>Compliant Solution</h2>
<pre>
let obj = {
  0:'a',
  name:'bob',
  1:'foo'
};

for (var property in obj) {
    if (obj.hasOwnProperty(property)) {
        writeln(obj[property]);  // outputs: a, foo, bob
    }
}
</pre>
