<p>The <code>void</code> operator evaluates it's argument and unconditionally returns <code>undefined</code>. There's no real justification for using
it. The assignment of <code>undefined</code> is a bad practice - it should only be seen as a starting value for an uninitialized variable - and the
evaluation of the argument can be handled by other means. Further, the use of <code>void</code>, which can be used with or without parentheses, so
could be confusing. </p>
<h2>Noncompliant Code Example</h2>
<pre>
void(function() {
   ...
}());
</pre>
<h2>Compliant Solution</h2>
<pre>
(function() {
   ...
}());
</pre>

