<p>The <code>render</code> should always return. The rule will raise an issue when itâ€™s missing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
const createReactClass = require('create-react-class');
const App1 = createReactClass({
  render() {
    &lt;div&gt;Contents&lt;/div&gt;; // Without return, it's an empty statement
  }
});

const React = require('react');
class App2 extends React.Component {
  render() {
    &lt;div&gt;Contents&lt;/div&gt;; // Without return, it's an empty statement
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
const createReactClass = require('create-react-class');
var App1 = createReactClass({
  render() {
    return &lt;div&gt;Contents&lt;/div&gt;;
  }
});

const React = require('react');
class App2 extends React.Component {
  render() {
    return &lt;div&gt;Contents&lt;/div&gt;;
  }
}
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://reactjs.org/docs/react-component.html#render">render()</a> - React API Reference </li>
</ul>

