<p>When writing the <code>render</code> method in a component it is easy to forget to return the JSX content. This rule will warn if the
<code>return</code> statement is missing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var Hello = createReactClass({
  render() {
    &lt;div&gt;Hello&lt;/div&gt;;
  }
});

class Hello extends React.Component {
  render() {
    &lt;div&gt;Hello&lt;/div&gt;;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
var Hello = createReactClass({
  render() {
    return &lt;div&gt;Hello&lt;/div&gt;;
  }
});

class Hello extends React.Component {
  render() {
    return &lt;div&gt;Hello&lt;/div&gt;;
  }
}
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-render-return.md">Enforce ES5 or ES6 class for
  returning value in render function (react/require-render-return) </a> - ESLint rule </li>
</ul>

