<h2>Why is this an issue?</h2>
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h3>Noncompliant code example</h3>
<p>With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
<pre>
function DoSomething(){...}  // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
function doSomething(){...}
</pre>
<h3>Exceptions</h3>
<p>This rule ignores React Functional Components, which are JavaScript functions named with a capital letter and returning a React element (JSX
syntax).</p>
<pre>
function Welcome() {
  const greeting = 'Hello, World!';

  // ...

  return (
    &lt;div className="Welcome"&gt;
      &lt;p&gt;{greeting}&lt;/p&gt;
    &lt;/div&gt;
  );
}
</pre>

