<h2>Why is this an issue?</h2>
<p>There are several reasons for a function not to have a function body:</p>
<ul>
  <li> It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production. </li>
  <li> It is not yet, or never will be, supported. In this case an exception should be thrown in languages where that mechanism is available. </li>
  <li> The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override. </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
function foo() {
}

var foo = () =&gt; {};
</pre>
<h3>Compliant solution</h3>
<pre>
function foo() {
    // This is intentional
}

var foo = () =&gt; {
    do_something();
};
</pre>
<h3>Exceptions</h3>
<ul>
  <li> This rule does not apply to function expressions and arrow functions as they can denote default values. </li>
</ul>
<pre>
static defaultProps = {
  listStyle: () =&gt; {}
};
</pre>
<ul>
  <li> The rule allows for empty functions with a name starting with the prefix <code>on</code> like <code>onClick</code>. </li>
</ul>
<pre>
function onClick() {
}
</pre>

