<h2>Why is this an issue?</h2>
<p>The use of wrapper objects for primitive types is gratuitous, confusing and dangerous. If you use a wrapper object constructor for type conversion,
just remove the <code>new</code> keyword, and youâ€™ll get a primitive value automatically. If you use a wrapper object as a way to add properties to a
primitive, you should re-think the design. Such uses are considered bad practice, and should be refactored.</p>
<h3>Noncompliant code example</h3>
<pre>
let x = new Number("0");
if (x) {
  alert('hi');  // Shows 'hi'.
}
</pre>
<h3>Compliant solution</h3>
<pre>
let x = Number("0");
if (x) {
  alert('hi');
}
</pre>

