<h2>Why is this an issue?</h2>
<p>Functions declared with the <code>set</code> keyword will automatically return the values they were passed. Thus any value explicitly returned from
a setter will be ignored, and explicitly returning a value is an error.</p>
<h3>Noncompliant code example</h3>
<pre>
var person = {
  // ...
  set name(name) {
    this.name = name;
    return 42;  // Noncompliant
  }
}
</pre>
<h3>Compliant solution</h3>
<pre>
var person = {
  // ...
  set name(name) {
    this.name = name;
  }
}
</pre>

