<h2>Why is this an issue?</h2>
<p>Empty character classes in regular expressions donâ€™t match anything.</p>
<p>This is likely a sign that the regular expression does not work as intended.</p>
<h3>Noncompliant code example</h3>
<pre>
/^foo[]/.test("foobar"); // false
</pre>
<h3>Compliant solution</h3>
<pre>
/^foo/.test("foobar"); // true
</pre>

