<h2>Why is this an issue?</h2>
<p>The <code>void</code> operator evaluates its argument and unconditionally returns <code>undefined</code>. It can be useful in pre-ECMAScript 5
environments, where <code>undefined</code> could be reassigned, but generally, its use makes code harder to understand.</p>
<h3>Noncompliant code example</h3>
<pre>
void doSomething();
</pre>
<h3>Compliant solution</h3>
<pre>
doSomething();
</pre>
<h3>Exceptions</h3>
<p>No issue is raised when <code>void 0</code> is used in place of <code>undefined</code>.</p>
<pre>
if (parameter === void 0) {...}
</pre>
<p>No issue is raised when <code>void</code> is used before immediately invoked function expressions.</p>
<pre>
void (function() {
   ...
}());
</pre>
<p>No issue is raised when <code>void</code>'s argument is a promise.</p>
<pre>
const promise = new Promise((resolve, reject) =&gt; {
   setTimeout(() =&gt; {
      resolve('done');
   }, 3000);
});
void promise;
</pre>

