<h2>Why is this an issue?</h2>
<p>The types of the arguments to built-in functions are specified in the JavaScript language specifications. Calls to these functions should conform
to the documented types, otherwise the result will most likely not be what was expected (e.g.: the call would always return <code>false</code>).</p>
<h3>Noncompliant code example</h3>
<pre>
const isTooSmall = Math.abs(x &lt; 0.0042);
</pre>
<h3>Compliant solution</h3>
<pre>
const isTooSmall = Math.abs(x) &lt; 0.0042;
</pre>

