<h2>Why is this an issue?</h2>
<p>Multiple imports from the same module should be merged together to improve readability.</p>
<h3>Noncompliant code example</h3>
<pre>
import { B1 } from 'b';
import { B2 } from 'b'; // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
import { B1, B2 } from 'b';
</pre>

