<h2>Why is this an issue?</h2>
<p>If you have an iterable, such as an array, set, or list, your best option for looping through its values is the <code>for of</code> syntax. Use a
counter, and …​ well you’ll get the right behavior, but your code just isn’t as clean or clear.</p>
<p>In a browser environment, <code>NodeList</code> and other array-like collections should work by default. If you are using TypeScript and seeing a
type error, make sure your configuration is correct.</p>
<h3>Noncompliant code example</h3>
<pre>
const arr = [4, 3, 2, 1];

for (let i = 0; i &lt; arr.length; i++) {  // Noncompliant
  console.log(arr[i]);
}
</pre>
<h3>Compliant solution</h3>
<pre>
const arr = [4, 3, 2, 1];

for (let value of arr) {
  console.log(value);
}
</pre>

