<h2>Why is this an issue?</h2>
<p>An array declared with missing ("sparse") elements is likely to be an error: an extra comma was inserted or perhaps the developer meant to insert
the missing value and forgot.</p>
<h3>Noncompliant code example</h3>
<pre>
let a = [1, , 3, 6, 9];  // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
let a = [1, 3, 6, 9];
</pre>

