<h2>Why is this an issue?</h2>
<p>Many assertions compare two objects or properties of these objects. Passing twice the same argument is likely to be a bug due to developerâ€™s
carelessness.</p>
<p>This rule raises an issue when a Chai assertion is given twice the same argument.</p>
<h3>Noncompliant code example</h3>
<pre>
const assert = require('chai').assert;

describe("test the same object", function() {
    it("uses chai 'assert'", function() {
        const expected = '1'
        const actual = (1).toString()
        assert.equal(actual, actual);  // Noncompliant
    });
});
</pre>
<h3>Compliant solution</h3>
<pre>
const assert = require('chai').assert;

describe("test the same object", function() {
    it("uses chai 'assert'", function() {
        const expected = '1'
        const actual = (1).toString()
        assert.equal(actual, expected);
    });
});
</pre>

