<h2>Why is this an issue?</h2>
<p>Regular expression literals should be preferred over the <code>RegExp</code> constructor calls when the pattern is a literal. Simply using a
regular expression literal is more concise and easier to read and does not require escaping like a string literal does.</p>
<p>Using the <code>RegExp</code> constructor is suitable when the pattern is computed dynamically, e.g. when it is provided by the user.</p>
<h3>Noncompliant code example</h3>
<pre>
new RegExp(/foo/);
new RegExp('bar');
new RegExp('baz', 'i');
new RegExp("\\d+");
new RegExp(`qux|quuz`);
</pre>
<h3>Compliant solution</h3>
<pre>
/foo/;
/bar/;
/baz/i;
/\d+/;
/qux|quuz/;
new RegExp(`Dear ${title},`);
</pre>

