<h2>Why is this an issue?</h2>
<p>React’s useState hook setter function should not be called directly in the body of a component, otherwise it would produce an infinite render loop.
This can happen by mistake. Most commonly, the setter function is called from an event handler.</p>
<h3>Noncompliant code example</h3>
<pre>
import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState("fr-FR");

    // Makes an infinite loop
    setLanguage(navigator.language);

    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage("fr-FR")}&gt;Je préfère le Français&lt;/button&gt;
      &lt;/section&gt;
    );
}
</pre>
<h3>Compliant solution</h3>
<pre>
import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState(navigator.language);

    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage("fr-FR")}&gt;Je préfère le Français&lt;/button&gt;
      &lt;/section&gt;
    );
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://reactjs.org/docs/hooks-state.html">hooks-state</a> - React API reference </li>
</ul>

