<p>Disallow the use of optional chaining in an expression where the <code>undefined</code> value would raise an error.</p>
<h2>Why is this an issue?</h2>
<p>The optional chaining operator <code>?.</code> allows to access a deeply nested property, returning <code>undefined</code> if the property or any
intermediate object is <code>undefined</code>.</p>
<p>This usually means that we expect the expression to evaluate as <code>undefined</code> in some cases. Therefore, using the optional chaining
operator in a context where returning <code>undefined</code> is forbidden can lead to errors.</p>
<h3>What is the potential impact?</h3>
<p>Since optional chaining represents multiple execution branches, having an error thrown in such a context can be hard to debug.</p>
<h2>How to fix it</h2>
<p>You should provide fallbacks for when the optional chaining operator is used to avoid raising errors.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre>
new (foo?.bar)();
const { foo } = bar?.baz;
const foo = [...bar?.baz]
</pre>
<h4>Compliant solution</h4>
<pre>
new (foo?.bar ?? baz)()
const { foo } = bar?.baz || {}
const foo = bar?.baz ? [...bar.baz] : []
</pre>

