<p>FIXME: add a description</p>
<h2>Why is this an issue?</h2>
<p>FIXME: remove the unused optional headers (that are commented out)</p>
<h2>How to fix it</h2>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre>
FIXME
</pre>
<h4>Compliant solution</h4>
<pre>
FIXME
</pre>

