<p>This rule prevents using the <code>void</code> type in ways that could be misleading.</p>
<p>You shouldnâ€™t use the void type outside of function returns and generics.</p>
<h2>Why is this an issue?</h2>
<p>The <code>void</code> type can only be mixed with <code>never</code> which can be quite blocking in TypeScript.</p>
<h2>How to fix it</h2>
<p>You should probably use <code>undefined</code> instead.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre>
type PossibleValues = string | number | void;
</pre>
<h4>Compliant solution</h4>
<pre>
type PossibleValues = string | number | undefined;
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://www.typescriptlang.org/docs/handbook/2/functions.html#void">TypeScript void type</a> </li>
  <li> <a href="https://www.typescriptlang.org/docs/handbook/2/functions.html#return-type-void">TypeScript void function return type</a> </li>
</ul>

