<h2>Why is this an issue?</h2>
<p>JavaScript allows any kind of value in boolean expressions. Using numbers, strings, and even objects as a boolean expression is possible. The
runtime coerces those to a boolean value. However, this automatic coercion can be misleading and lead to unexpected results.</p>
<p>The rule forbids the usage of strings and numbers in expressions where a boolean is expected:</p>
<ul>
  <li> Argument to the logical negation operator (<code>!arg</code>). </li>
  <li> The condition in a conditional expression (<code>cond ? x : y</code>). </li>
  <li> Conditions for <code>if</code>, <code>for</code>, <code>while</code>, and <code>do-while</code> statements. </li>
  <li> Operands of logical binary operators (<code>||</code>, <code>&amp;&amp;</code>). </li>
</ul>
<p>The TSConfig needs to set <code>strictNullChecks</code> to <code>true</code> for the rule to work properly.</p>
<h3>What is the potential impact?</h3>
<p>Numbers get coerced to <code>true</code> except for the zero and <code>NaN</code> which are converted to <code>false</code>.</p>
<p>Strings get coerced to <code>true</code> as well, except for the empty string <code>''</code> which is converted to <code>false</code>.</p>
<p>These special cases can lead to bugs.</p>
<h2>How to fix it</h2>
<p>Either check the string or numberâ€™s value or use the function constructor <code>Boolean</code> to convert it to boolean.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre>
const x = [1,2,3];
if (x.length) {
    /* ... */
}
</pre>
<h4>Compliant solution</h4>
<pre>
const x = [1,2,3];
if (x.length &gt; 0) {
    /* ... */
}
</pre>
<h4>Noncompliant code example</h4>
<pre>
const s = 'foo';
if (s) {
    /* ... */
}
</pre>
<h4>Compliant solution</h4>
<pre>
const s = 'foo';
if (s !== '') {
    /* ... */
}
</pre>
<h4>Noncompliant code example</h4>
<pre>
const b = BigInt(42);
if (b) {
    /* ... */
}
</pre>
<h4>Compliant solution</h4>
<pre>
const b = BigInt(42);
if (Boolean(b)) {
    /* ... */
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion">MDN Boolean coercion</a>
  </li>
</ul>

